{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 0 Loading packages <a name=\"loading\"></a> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home2/s418610/.conda/envs/py37_res_PAGA/bin/python\n",
      "['/home2/s418610/jupyter_notebooks/03.Invivo_neuronal_reprogramming/Manuscript/to_GitHub', '', '/cm/shared/apps/cnvkit/0.9.5/lib/python3.6/site-packages', '/cm/shared/apps/python/3.6.4-anaconda/lib', '/cm/shared/apps/jupyterkernel-matlab/0.16.1/lib/python3.6/site-packages', '/cm/shared/apps/python/2.7.x-anaconda/lib', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python37.zip', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7/lib-dynload', '/home2/s418610/.local/lib/python3.7/site-packages', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7/site-packages', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7/site-packages/fa2-0.3.5-py3.7-linux-x86_64.egg', '/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7/site-packages/IPython/extensions', '/home2/s418610/.ipython']\n",
      "3.7.3 (default, Mar 27 2019, 22:11:17) \n",
      "[GCC 7.3.0]\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "print(sys.executable)\n",
    "print(sys.path)\n",
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from pathlib import Path\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scipy.sparse\n",
    "import sklearn.manifold\n",
    "import statsmodels.api as sm\n",
    "\n",
    "# from sklearn.preprocessing import StandardScaler\n",
    "# from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE\n",
    "from pyclustering.cluster.kmedoids import kmedoids\n",
    "\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "# import seaborn as sb\n",
    "\n",
    "import scanpy as sc\n",
    "import harmonypy as hm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "numpy 1.17.2\n",
      "pandas 0.25.2\n",
      "scipy 1.3.1\n",
      "sklearn 0.21.3\n",
      "statsmodels 0.10.1\n",
      "matplotlib 3.3.2\n",
      "seaborn 0.9.0\n",
      "scanpy==1.4.5.post1 anndata==0.6.22.post1 umap==0.3.10 numpy==1.17.2 scipy==1.3.1 pandas==0.25.2 scikit-learn==0.21.3 statsmodels==0.10.1 python-igraph==0.7.1 louvain==0.6.1\n",
      "2021-03-05\n"
     ]
    }
   ],
   "source": [
    "plt.rcParams['figure.figsize']=(8,8) #rescale figures\n",
    "sc.settings.verbosity = 3\n",
    "\n",
    "print('numpy', np.__version__)\n",
    "print('pandas', pd.__version__)\n",
    "print('scipy', scipy.__version__)\n",
    "print('sklearn', sklearn.__version__)\n",
    "print('statsmodels', sm.__version__)\n",
    "print('matplotlib', mpl.__version__)\n",
    "print('seaborn', sns.__version__)\n",
    "\n",
    "sc.logging.print_versions()\n",
    "\n",
    "!date +%F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import anndata\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import statsmodels.api as sm\n",
    "lowess = sm.nonparametric.lowess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import chi2_contingency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib_venn import venn2, venn2_circles, venn2_unweighted\n",
    "def intersection(lst1, lst2): \n",
    "    return list(set(lst1) & set(lst2)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# set a working directory for saving plots\n",
    "wdir = \"/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/99.Manuscript/Figures_v2\"\n",
    "# wdir = \"/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/99.Manuscript/Figures/Figure2/\"\n",
    "os.chdir( wdir )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# A sub-directory for saving intermediate data for organized pipeline v1\n",
    "data_dir = '/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/analysis/Cluster_with_10X_20k/data/10x_remapped/PipelineOrganized_v1/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "neural_data_dir = data_dir + 'NeuralOnly/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import statsmodels.api as sm\n",
    "lowess = sm.nonparametric.lowess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import chi2_contingency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib_venn import venn2, venn2_circles, venn2_unweighted\n",
    "def intersection(lst1, lst2): \n",
    "    return list(set(lst1) & set(lst2)) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1 Load data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_data_dir = '/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/analysis/Cluster_with_10X_20k/data/10x_remapped/PipelineOrganized_v1/Gene_clusters/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_path = gene_data_dir + 'adata_genes_Manuscript_v2'\n",
    "with open(input_path, 'rb') as input_file:\n",
    " \n",
    "    adata_genes = pickle.load(input_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 2469 × 4601 \n",
       "    obs: 'gene_ids', 'feature_types', 'genome', 'n_cells', 'p_val_adj', 'n_counts', 'n_cells_experimental', 'louvain', 'log10_n_cells_experimental'\n",
       "    var: 'library_name', 'batch', 'lenti_count', 'lenti_log1p_count', 'tdT_count', 'tdT_log1p_count', 'n_counts', 'log_counts', 'n_genes', 'mt_count', 'mt_frac', 'n_counts_all', 'louvain', 'dpt_pseudotime', 'rank_pseudotime', 'Rpl32', 'Egfr'\n",
       "    uns: 'pca', 'neighbors', 'louvain', 'paga', 'louvain_sizes', 'louvain_colors', 'umap'\n",
       "    obsm: 'X_pca', 'X_umap'\n",
       "    varm: 'PCs'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     582\n",
       "1     313\n",
       "2     309\n",
       "3     302\n",
       "4     240\n",
       "5     218\n",
       "6     135\n",
       "7     131\n",
       "8      91\n",
       "9      76\n",
       "10     56\n",
       "11     16\n",
       "Name: louvain, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# How many genes are in each cluster? \n",
    "louvain_clusters_counter = adata_genes.obs['louvain'].value_counts()\n",
    "louvain_clusters_counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "      <th>n_cells</th>\n",
       "      <th>p_val_adj</th>\n",
       "      <th>n_counts</th>\n",
       "      <th>n_cells_experimental</th>\n",
       "      <th>louvain</th>\n",
       "      <th>log10_n_cells_experimental</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Rgs20</th>\n",
       "      <td>ENSMUSG00000002459</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>1062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>951.583130</td>\n",
       "      <td>1062</td>\n",
       "      <td>0</td>\n",
       "      <td>3.026533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>St18</th>\n",
       "      <td>ENSMUSG00000033740</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>276</td>\n",
       "      <td>NaN</td>\n",
       "      <td>239.741074</td>\n",
       "      <td>276</td>\n",
       "      <td>2</td>\n",
       "      <td>2.442480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rrs1</th>\n",
       "      <td>ENSMUSG00000061024</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>806.224243</td>\n",
       "      <td>966</td>\n",
       "      <td>3</td>\n",
       "      <td>2.985426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adhfe1</th>\n",
       "      <td>ENSMUSG00000025911</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>547</td>\n",
       "      <td>NaN</td>\n",
       "      <td>427.636810</td>\n",
       "      <td>547</td>\n",
       "      <td>0</td>\n",
       "      <td>2.738781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mybl1</th>\n",
       "      <td>ENSMUSG00000025912</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>109.949699</td>\n",
       "      <td>140</td>\n",
       "      <td>1</td>\n",
       "      <td>2.149219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Eif3a</th>\n",
       "      <td>ENSMUSG00000024991</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>2458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2743.859375</td>\n",
       "      <td>2257</td>\n",
       "      <td>1</td>\n",
       "      <td>3.353724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mt-Nd4l</th>\n",
       "      <td>ENSMUSG00000065947</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>2930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3241.866699</td>\n",
       "      <td>2022</td>\n",
       "      <td>0</td>\n",
       "      <td>3.305996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mt-Nd5</th>\n",
       "      <td>ENSMUSG00000064367</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>3469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5341.198242</td>\n",
       "      <td>1945</td>\n",
       "      <td>0</td>\n",
       "      <td>3.289143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lenti-hGfa-Dlx2-IRES-GFP</th>\n",
       "      <td>Lenti-hGfa-Dlx2-IRES-GFP</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>2411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3311.665039</td>\n",
       "      <td>2140</td>\n",
       "      <td>7</td>\n",
       "      <td>3.330617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tdTomato-WPRE-bGHpA</th>\n",
       "      <td>tdTomato-WPRE-bGHpA</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0-Lenti-Dlx2_tdTomato</td>\n",
       "      <td>3115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4062.851562</td>\n",
       "      <td>2421</td>\n",
       "      <td>3</td>\n",
       "      <td>3.384174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2469 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          gene_ids    feature_types  \\\n",
       "Rgs20                           ENSMUSG00000002459  Gene Expression   \n",
       "St18                            ENSMUSG00000033740  Gene Expression   \n",
       "Rrs1                            ENSMUSG00000061024  Gene Expression   \n",
       "Adhfe1                          ENSMUSG00000025911  Gene Expression   \n",
       "Mybl1                           ENSMUSG00000025912  Gene Expression   \n",
       "...                                            ...              ...   \n",
       "Eif3a                           ENSMUSG00000024991  Gene Expression   \n",
       "mt-Nd4l                         ENSMUSG00000065947  Gene Expression   \n",
       "mt-Nd5                          ENSMUSG00000064367  Gene Expression   \n",
       "Lenti-hGfa-Dlx2-IRES-GFP  Lenti-hGfa-Dlx2-IRES-GFP  Gene Expression   \n",
       "tdTomato-WPRE-bGHpA            tdTomato-WPRE-bGHpA  Gene Expression   \n",
       "\n",
       "                                                  genome  n_cells  p_val_adj  \\\n",
       "Rgs20                     mm10-3.0.0-Lenti-Dlx2_tdTomato     1062        NaN   \n",
       "St18                      mm10-3.0.0-Lenti-Dlx2_tdTomato      276        NaN   \n",
       "Rrs1                      mm10-3.0.0-Lenti-Dlx2_tdTomato      966        NaN   \n",
       "Adhfe1                    mm10-3.0.0-Lenti-Dlx2_tdTomato      547        NaN   \n",
       "Mybl1                     mm10-3.0.0-Lenti-Dlx2_tdTomato      140        NaN   \n",
       "...                                                  ...      ...        ...   \n",
       "Eif3a                     mm10-3.0.0-Lenti-Dlx2_tdTomato     2458        NaN   \n",
       "mt-Nd4l                   mm10-3.0.0-Lenti-Dlx2_tdTomato     2930        NaN   \n",
       "mt-Nd5                    mm10-3.0.0-Lenti-Dlx2_tdTomato     3469        NaN   \n",
       "Lenti-hGfa-Dlx2-IRES-GFP  mm10-3.0.0-Lenti-Dlx2_tdTomato     2411        NaN   \n",
       "tdTomato-WPRE-bGHpA       mm10-3.0.0-Lenti-Dlx2_tdTomato     3115        NaN   \n",
       "\n",
       "                             n_counts  n_cells_experimental louvain  \\\n",
       "Rgs20                      951.583130                  1062       0   \n",
       "St18                       239.741074                   276       2   \n",
       "Rrs1                       806.224243                   966       3   \n",
       "Adhfe1                     427.636810                   547       0   \n",
       "Mybl1                      109.949699                   140       1   \n",
       "...                               ...                   ...     ...   \n",
       "Eif3a                     2743.859375                  2257       1   \n",
       "mt-Nd4l                   3241.866699                  2022       0   \n",
       "mt-Nd5                    5341.198242                  1945       0   \n",
       "Lenti-hGfa-Dlx2-IRES-GFP  3311.665039                  2140       7   \n",
       "tdTomato-WPRE-bGHpA       4062.851562                  2421       3   \n",
       "\n",
       "                          log10_n_cells_experimental  \n",
       "Rgs20                                       3.026533  \n",
       "St18                                        2.442480  \n",
       "Rrs1                                        2.985426  \n",
       "Adhfe1                                      2.738781  \n",
       "Mybl1                                       2.149219  \n",
       "...                                              ...  \n",
       "Eif3a                                       3.353724  \n",
       "mt-Nd4l                                     3.305996  \n",
       "mt-Nd5                                      3.289143  \n",
       "Lenti-hGfa-Dlx2-IRES-GFP                    3.330617  \n",
       "tdTomato-WPRE-bGHpA                         3.384174  \n",
       "\n",
       "[2469 rows x 9 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_genes.obs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2 Compare genes from Lindtner et al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_overlap = pd.DataFrame(index=adata_genes.obs['louvain'].cat.categories, columns=['num_genes_in_cluster', 'down_genes', 'up_genes', 'aRE_genes', 'rRE_genes'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_genes_in_cluster</th>\n",
       "      <th>down_genes</th>\n",
       "      <th>up_genes</th>\n",
       "      <th>aRE_genes</th>\n",
       "      <th>rRE_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   num_genes_in_cluster down_genes up_genes aRE_genes rRE_genes\n",
       "0                   NaN        NaN      NaN       NaN       NaN\n",
       "1                   NaN        NaN      NaN       NaN       NaN\n",
       "2                   NaN        NaN      NaN       NaN       NaN\n",
       "3                   NaN        NaN      NaN       NaN       NaN\n",
       "4                   NaN        NaN      NaN       NaN       NaN\n",
       "5                   NaN        NaN      NaN       NaN       NaN\n",
       "6                   NaN        NaN      NaN       NaN       NaN\n",
       "7                   NaN        NaN      NaN       NaN       NaN\n",
       "8                   NaN        NaN      NaN       NaN       NaN\n",
       "9                   NaN        NaN      NaN       NaN       NaN\n",
       "10                  NaN        NaN      NaN       NaN       NaN\n",
       "11                  NaN        NaN      NaN       NaN       NaN"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.1 Compare with Lindtner et al. DEGs in DLX KO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "Lindtner_genes = pd.read_csv('/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/analysis/10.Comparison_w_Lindtner2019/DE_genes/DE_genes_Lindtner2019.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "down_genes = Lindtner_genes['Gene'][(Lindtner_genes['logFC'] < 0).tolist()].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_genes = Lindtner_genes['Gene'][(Lindtner_genes['logFC'] > 0).tolist()].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "181"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(down_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(up_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1.1 Down genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "for current_cluster in adata_genes.obs['louvain'].cat.categories: \n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    df_overlap.loc[current_cluster, 'down_genes'] = len(intersection(gene_cluster, down_genes))\n",
    "    df_overlap.loc[current_cluster, 'num_genes_in_cluster'] = len(gene_cluster)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1.2 Up genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "for current_cluster in adata_genes.obs['louvain'].cat.categories: \n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    df_overlap.loc[current_cluster, 'up_genes'] = len(intersection(gene_cluster, up_genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.2 Compare with Lindtner et al. aRE- and rRE-associated genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home2/s418610/.conda/envs/py37_res_PAGA/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (18,20,121,123,124,125,126,128) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "Lindtner_peaks = pd.read_csv('/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/analysis/10.Comparison_w_Lindtner2019/REs/Dlx_ChIP-seq_peaks.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>PeakID</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>nearest.TSS</th>\n",
       "      <th>dist.nearest.TSS</th>\n",
       "      <th>Feature</th>\n",
       "      <th>VertebratePhastcons</th>\n",
       "      <th>...</th>\n",
       "      <th>LGE.MZ</th>\n",
       "      <th>MGE.VZ</th>\n",
       "      <th>MGE.SVZ</th>\n",
       "      <th>MGE.MZ</th>\n",
       "      <th>POA.VZ</th>\n",
       "      <th>POA.SVZ</th>\n",
       "      <th>POA.MZ</th>\n",
       "      <th>Se.VZ</th>\n",
       "      <th>Se.SVZ</th>\n",
       "      <th>Se.MZ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3084850</td>\n",
       "      <td>3085335</td>\n",
       "      <td>486</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>120376</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>262.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3200383</td>\n",
       "      <td>3200701</td>\n",
       "      <td>319</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>5010</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>470.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3279440</td>\n",
       "      <td>3279901</td>\n",
       "      <td>462</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>73726</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>272.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Rp1</td>\n",
       "      <td>Merged.4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4322611</td>\n",
       "      <td>4322865</td>\n",
       "      <td>255</td>\n",
       "      <td>Rp1_RefSeq</td>\n",
       "      <td>27528</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mrpl15</td>\n",
       "      <td>Merged.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4775680</td>\n",
       "      <td>4776335</td>\n",
       "      <td>656</td>\n",
       "      <td>Mrpl15_RefSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>250.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16914</th>\n",
       "      <td>Tlr7</td>\n",
       "      <td>Merged.17127</td>\n",
       "      <td>chrX</td>\n",
       "      <td>163759141</td>\n",
       "      <td>163759683</td>\n",
       "      <td>543</td>\n",
       "      <td>Tlr7_UCSC</td>\n",
       "      <td>8346</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>330.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16915</th>\n",
       "      <td>Frmpd4</td>\n",
       "      <td>Merged.17128</td>\n",
       "      <td>chrX</td>\n",
       "      <td>164211555</td>\n",
       "      <td>164211920</td>\n",
       "      <td>366</td>\n",
       "      <td>Frmpd4_RefSeq</td>\n",
       "      <td>81087</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>285.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16916</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>Merged.17129</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166275796</td>\n",
       "      <td>166276628</td>\n",
       "      <td>833</td>\n",
       "      <td>Mid1_RefSeq</td>\n",
       "      <td>9704</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>303.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16917</th>\n",
       "      <td>Fxy</td>\n",
       "      <td>Merged.17130</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166407669</td>\n",
       "      <td>166414496</td>\n",
       "      <td>6828</td>\n",
       "      <td>Fxy_UCSC</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>292.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16918</th>\n",
       "      <td>G530011O06Rik</td>\n",
       "      <td>Merged.17131</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166414737</td>\n",
       "      <td>166423349</td>\n",
       "      <td>8613</td>\n",
       "      <td>G530011O06Rik_RefSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>256.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16919 rows × 135 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene        PeakID seqnames      start        end  width  \\\n",
       "0          mKIAA1889      Merged.1     chr1    3084850    3085335    486   \n",
       "1          mKIAA1889      Merged.2     chr1    3200383    3200701    319   \n",
       "2          mKIAA1889      Merged.3     chr1    3279440    3279901    462   \n",
       "3                Rp1      Merged.4     chr1    4322611    4322865    255   \n",
       "4             Mrpl15      Merged.5     chr1    4775680    4776335    656   \n",
       "...              ...           ...      ...        ...        ...    ...   \n",
       "16914           Tlr7  Merged.17127     chrX  163759141  163759683    543   \n",
       "16915         Frmpd4  Merged.17128     chrX  164211555  164211920    366   \n",
       "16916           Mid1  Merged.17129     chrX  166275796  166276628    833   \n",
       "16917            Fxy  Merged.17130     chrX  166407669  166414496   6828   \n",
       "16918  G530011O06Rik  Merged.17131     chrX  166414737  166423349   8613   \n",
       "\n",
       "                nearest.TSS  dist.nearest.TSS     Feature  \\\n",
       "0            mKIAA1889_UCSC            120376  Intergenic   \n",
       "1            mKIAA1889_UCSC              5010    GeneBody   \n",
       "2            mKIAA1889_UCSC             73726    GeneBody   \n",
       "3                Rp1_RefSeq             27528    GeneBody   \n",
       "4             Mrpl15_RefSeq                 0         TSS   \n",
       "...                     ...               ...         ...   \n",
       "16914             Tlr7_UCSC              8346    GeneBody   \n",
       "16915         Frmpd4_RefSeq             81087    GeneBody   \n",
       "16916           Mid1_RefSeq              9704    GeneBody   \n",
       "16917              Fxy_UCSC                 0         TSS   \n",
       "16918  G530011O06Rik_RefSeq                 0         TSS   \n",
       "\n",
       "       VertebratePhastcons  ...  LGE.MZ  MGE.VZ  MGE.SVZ MGE.MZ POA.VZ  \\\n",
       "0                    262.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "1                    470.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "2                    272.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "3                      NaN  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "4                    250.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "...                    ...  ...     ...     ...      ...    ...    ...   \n",
       "16914                330.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16915                285.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16916                303.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16917                292.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16918                256.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "\n",
       "      POA.SVZ  POA.MZ  Se.VZ Se.SVZ Se.MZ  \n",
       "0         NaN     NaN    NaN    NaN   NaN  \n",
       "1         NaN     NaN    NaN    NaN   NaN  \n",
       "2         NaN     NaN    NaN    NaN   NaN  \n",
       "3         NaN     NaN    NaN    NaN   NaN  \n",
       "4         NaN     NaN    NaN    NaN   NaN  \n",
       "...       ...     ...    ...    ...   ...  \n",
       "16914     NaN     NaN    NaN    NaN   NaN  \n",
       "16915     NaN     NaN    NaN    NaN   NaN  \n",
       "16916     NaN     NaN    NaN    NaN   NaN  \n",
       "16917     NaN     NaN    NaN    NaN   NaN  \n",
       "16918     NaN     NaN    NaN    NaN   NaN  \n",
       "\n",
       "[16919 rows x 135 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Lindtner_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "aRE_regions = Lindtner_peaks[Lindtner_peaks['a.RE.bdg'] == 1]\n",
    "\n",
    "aRE_genes = aRE_regions['Gene']\n",
    "\n",
    "aRE_genes = aRE_genes.unique().tolist()\n",
    "\n",
    "rRE_regions = Lindtner_peaks[Lindtner_peaks['r.RE.bdg'] == 1]\n",
    "\n",
    "rRE_genes = rRE_regions['Gene']\n",
    "\n",
    "rRE_genes = rRE_genes.unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "447"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(aRE_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "418"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rRE_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2.1 aRE-associated genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "for current_cluster in adata_genes.obs['louvain'].cat.categories: \n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    df_overlap.loc[current_cluster, 'aRE_genes'] = len(intersection(gene_cluster, aRE_genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2.2 rRE-associated genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "for current_cluster in adata_genes.obs['louvain'].cat.categories: \n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    df_overlap.loc[current_cluster, 'rRE_genes'] = len(intersection(gene_cluster, rRE_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_genes_in_cluster</th>\n",
       "      <th>down_genes</th>\n",
       "      <th>up_genes</th>\n",
       "      <th>aRE_genes</th>\n",
       "      <th>rRE_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>582</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>313</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>309</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>302</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>240</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>218</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>131</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>76</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_genes_in_cluster  down_genes  up_genes  aRE_genes  rRE_genes\n",
       "0                    582           2        14         10         18\n",
       "1                    313           2         7          5         13\n",
       "2                    309           7         3         11         13\n",
       "3                    302           0         1          5         11\n",
       "4                    240          24         1         27         20\n",
       "5                    218           2         1          5          7\n",
       "6                    135           1         0          2          0\n",
       "7                    131           5         2          3          4\n",
       "8                     91           0         0          0          1\n",
       "9                     76           1         1          0          1\n",
       "10                    56           0         0          0          0\n",
       "11                    16           0         0          0          1"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_overlap.to_csv('Overlap_summary.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.3 Compare with Lindtner et al. DLX-bound genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "Lindtner_peaks = pd.read_csv('/project/GCRB/Hon_lab/s418610/Projects/03.Invivo_neuronal_reprogramming/analysis/10.Comparison_w_Lindtner2019/REs/Dlx_ChIP-seq_peaks.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>PeakID</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>nearest.TSS</th>\n",
       "      <th>dist.nearest.TSS</th>\n",
       "      <th>Feature</th>\n",
       "      <th>VertebratePhastcons</th>\n",
       "      <th>...</th>\n",
       "      <th>LGE.MZ</th>\n",
       "      <th>MGE.VZ</th>\n",
       "      <th>MGE.SVZ</th>\n",
       "      <th>MGE.MZ</th>\n",
       "      <th>POA.VZ</th>\n",
       "      <th>POA.SVZ</th>\n",
       "      <th>POA.MZ</th>\n",
       "      <th>Se.VZ</th>\n",
       "      <th>Se.SVZ</th>\n",
       "      <th>Se.MZ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3084850</td>\n",
       "      <td>3085335</td>\n",
       "      <td>486</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>120376</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>262.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3200383</td>\n",
       "      <td>3200701</td>\n",
       "      <td>319</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>5010</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>470.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mKIAA1889</td>\n",
       "      <td>Merged.3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3279440</td>\n",
       "      <td>3279901</td>\n",
       "      <td>462</td>\n",
       "      <td>mKIAA1889_UCSC</td>\n",
       "      <td>73726</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>272.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Rp1</td>\n",
       "      <td>Merged.4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4322611</td>\n",
       "      <td>4322865</td>\n",
       "      <td>255</td>\n",
       "      <td>Rp1_RefSeq</td>\n",
       "      <td>27528</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mrpl15</td>\n",
       "      <td>Merged.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4775680</td>\n",
       "      <td>4776335</td>\n",
       "      <td>656</td>\n",
       "      <td>Mrpl15_RefSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>250.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16914</th>\n",
       "      <td>Tlr7</td>\n",
       "      <td>Merged.17127</td>\n",
       "      <td>chrX</td>\n",
       "      <td>163759141</td>\n",
       "      <td>163759683</td>\n",
       "      <td>543</td>\n",
       "      <td>Tlr7_UCSC</td>\n",
       "      <td>8346</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>330.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16915</th>\n",
       "      <td>Frmpd4</td>\n",
       "      <td>Merged.17128</td>\n",
       "      <td>chrX</td>\n",
       "      <td>164211555</td>\n",
       "      <td>164211920</td>\n",
       "      <td>366</td>\n",
       "      <td>Frmpd4_RefSeq</td>\n",
       "      <td>81087</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>285.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16916</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>Merged.17129</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166275796</td>\n",
       "      <td>166276628</td>\n",
       "      <td>833</td>\n",
       "      <td>Mid1_RefSeq</td>\n",
       "      <td>9704</td>\n",
       "      <td>GeneBody</td>\n",
       "      <td>303.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16917</th>\n",
       "      <td>Fxy</td>\n",
       "      <td>Merged.17130</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166407669</td>\n",
       "      <td>166414496</td>\n",
       "      <td>6828</td>\n",
       "      <td>Fxy_UCSC</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>292.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16918</th>\n",
       "      <td>G530011O06Rik</td>\n",
       "      <td>Merged.17131</td>\n",
       "      <td>chrX</td>\n",
       "      <td>166414737</td>\n",
       "      <td>166423349</td>\n",
       "      <td>8613</td>\n",
       "      <td>G530011O06Rik_RefSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>TSS</td>\n",
       "      <td>256.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16919 rows × 135 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene        PeakID seqnames      start        end  width  \\\n",
       "0          mKIAA1889      Merged.1     chr1    3084850    3085335    486   \n",
       "1          mKIAA1889      Merged.2     chr1    3200383    3200701    319   \n",
       "2          mKIAA1889      Merged.3     chr1    3279440    3279901    462   \n",
       "3                Rp1      Merged.4     chr1    4322611    4322865    255   \n",
       "4             Mrpl15      Merged.5     chr1    4775680    4776335    656   \n",
       "...              ...           ...      ...        ...        ...    ...   \n",
       "16914           Tlr7  Merged.17127     chrX  163759141  163759683    543   \n",
       "16915         Frmpd4  Merged.17128     chrX  164211555  164211920    366   \n",
       "16916           Mid1  Merged.17129     chrX  166275796  166276628    833   \n",
       "16917            Fxy  Merged.17130     chrX  166407669  166414496   6828   \n",
       "16918  G530011O06Rik  Merged.17131     chrX  166414737  166423349   8613   \n",
       "\n",
       "                nearest.TSS  dist.nearest.TSS     Feature  \\\n",
       "0            mKIAA1889_UCSC            120376  Intergenic   \n",
       "1            mKIAA1889_UCSC              5010    GeneBody   \n",
       "2            mKIAA1889_UCSC             73726    GeneBody   \n",
       "3                Rp1_RefSeq             27528    GeneBody   \n",
       "4             Mrpl15_RefSeq                 0         TSS   \n",
       "...                     ...               ...         ...   \n",
       "16914             Tlr7_UCSC              8346    GeneBody   \n",
       "16915         Frmpd4_RefSeq             81087    GeneBody   \n",
       "16916           Mid1_RefSeq              9704    GeneBody   \n",
       "16917              Fxy_UCSC                 0         TSS   \n",
       "16918  G530011O06Rik_RefSeq                 0         TSS   \n",
       "\n",
       "       VertebratePhastcons  ...  LGE.MZ  MGE.VZ  MGE.SVZ MGE.MZ POA.VZ  \\\n",
       "0                    262.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "1                    470.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "2                    272.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "3                      NaN  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "4                    250.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "...                    ...  ...     ...     ...      ...    ...    ...   \n",
       "16914                330.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16915                285.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16916                303.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16917                292.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "16918                256.0  ...     NaN     NaN      NaN    NaN    NaN   \n",
       "\n",
       "      POA.SVZ  POA.MZ  Se.VZ Se.SVZ Se.MZ  \n",
       "0         NaN     NaN    NaN    NaN   NaN  \n",
       "1         NaN     NaN    NaN    NaN   NaN  \n",
       "2         NaN     NaN    NaN    NaN   NaN  \n",
       "3         NaN     NaN    NaN    NaN   NaN  \n",
       "4         NaN     NaN    NaN    NaN   NaN  \n",
       "...       ...     ...    ...    ...   ...  \n",
       "16914     NaN     NaN    NaN    NaN   NaN  \n",
       "16915     NaN     NaN    NaN    NaN   NaN  \n",
       "16916     NaN     NaN    NaN    NaN   NaN  \n",
       "16917     NaN     NaN    NaN    NaN   NaN  \n",
       "16918     NaN     NaN    NaN    NaN   NaN  \n",
       "\n",
       "[16919 rows x 135 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Lindtner_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_fraction_overlap = df_overlap.T/adata_genes.obs['louvain'].value_counts().values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_fraction_overlap = df_fraction_overlap.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>down_genes</th>\n",
       "      <th>up_genes</th>\n",
       "      <th>aRE_genes</th>\n",
       "      <th>rRE_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.003436</td>\n",
       "      <td>0.024055</td>\n",
       "      <td>0.017182</td>\n",
       "      <td>0.030928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.006390</td>\n",
       "      <td>0.022364</td>\n",
       "      <td>0.015974</td>\n",
       "      <td>0.041534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.022654</td>\n",
       "      <td>0.009709</td>\n",
       "      <td>0.035599</td>\n",
       "      <td>0.042071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003311</td>\n",
       "      <td>0.016556</td>\n",
       "      <td>0.036424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.004167</td>\n",
       "      <td>0.112500</td>\n",
       "      <td>0.083333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.009174</td>\n",
       "      <td>0.004587</td>\n",
       "      <td>0.022936</td>\n",
       "      <td>0.032110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.007407</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014815</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.038168</td>\n",
       "      <td>0.015267</td>\n",
       "      <td>0.022901</td>\n",
       "      <td>0.030534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.013158</td>\n",
       "      <td>0.013158</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.013158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    down_genes  up_genes  aRE_genes  rRE_genes\n",
       "0     0.003436  0.024055   0.017182   0.030928\n",
       "1     0.006390  0.022364   0.015974   0.041534\n",
       "2     0.022654  0.009709   0.035599   0.042071\n",
       "3     0.000000  0.003311   0.016556   0.036424\n",
       "4     0.100000  0.004167   0.112500   0.083333\n",
       "5     0.009174  0.004587   0.022936   0.032110\n",
       "6     0.007407  0.000000   0.014815   0.000000\n",
       "7     0.038168  0.015267   0.022901   0.030534\n",
       "8     0.000000  0.000000   0.000000   0.010989\n",
       "9     0.013158  0.013158   0.000000   0.013158\n",
       "10    0.000000  0.000000   0.000000   0.000000\n",
       "11    0.000000  0.000000   0.000000   0.062500"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_fraction_overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3 Print important overlap genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "# current_cluster = '4'\n",
    "for current_cluster in range(12):\n",
    "    current_cluster = str(current_cluster)\n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    overlap_down = intersection(gene_cluster, down_genes)\n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    overlap_up = intersection(gene_cluster, up_genes)\n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    overlap_aRE = intersection(gene_cluster, aRE_genes)\n",
    "\n",
    "    gene_cluster = adata_genes.obs.loc[adata_genes.obs['louvain']==current_cluster].index\n",
    "\n",
    "    overlap_rRE = intersection(gene_cluster, rRE_genes)\n",
    "\n",
    "    overlap_down = pd.DataFrame(data=overlap_down)\n",
    "\n",
    "    overlap_down.to_csv('Cluster'+current_cluster+'Genes_overlap_down.csv')\n",
    "\n",
    "    overlap_up = pd.DataFrame(data=overlap_up)\n",
    "\n",
    "    overlap_up.to_csv('Cluster'+current_cluster+'Genes_overlap_up.csv')\n",
    "\n",
    "    overlap_aRE = pd.DataFrame(data=overlap_aRE)\n",
    "\n",
    "    overlap_aRE.to_csv('Cluster'+current_cluster+'Genes_overlap_aRE.csv')\n",
    "\n",
    "    overlap_rRE = pd.DataFrame(data=overlap_rRE)\n",
    "\n",
    "    overlap_rRE.to_csv('Cluster'+current_cluster+'Genes_overlap_rRE.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py37_res_PAGA",
   "language": "python",
   "name": "py37_res_paga"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
